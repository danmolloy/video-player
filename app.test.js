import { act } from "react-dom/test-utils"
import { unmountComponentAtNode, render, createRoot } from "react-dom";
import Home from './pages/index.tsx'
let container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
})

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
})

describe("App", () => {
  it("Loads", () => {
    act(() => {
      render(<Home />, container)
    })
    
    expect(container.innerHTML).toMatchSnapshot()
  })

  it("Plays multiple videos sequentially", () => {})
  it("Has error page", () => {})
  it("Shows title of current video", () => {})
  it("Has nightmode", () => {})
  it("Has about page", () => {})
})

describe("Video Player", () => {
  it("Shows loading indicator based on state of video player", async () => {
    act(() => {
      render(<Home />, container)
    })
    await act(async () => {
      await new Promise(res => setTimeout(res, 1000))
    })
    console.log(container.textContent)
  })

  it("Has play/pause button", () => {})
  it("Has volume button", () => {})
  it("Has progress bar with seeking feature", () => {})
  it("Has full screen toggle", () => {})
  it("Shows elapsed time and total time in HH:MM", () => {})
  it("Playback speed control", () => {})
  it("Next video button", () => {})
})


describe("Playlist", () => {
  it("Displays file names rather than full URL", () => {})
  it("Highlights current selected movie", () => {})
  it("Capitalize each word of fileName", () => {})
  it("If final video ends, plays first video", () => {})
})