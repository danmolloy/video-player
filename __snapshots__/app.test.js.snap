// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Loads 1`] = `"<div class=\\"index-div\\"><div class=\\"title-video-player-div\\"><h1 class=\\"title\\">Master</h1><div class=\\"video-player-div\\"><video id=\\"video\\" width=\\"640\\" height=\\"480\\"></video><div class=\\"vid-controls-div\\"><div class=\\"progress-div\\"><div class=\\"loading-wheel\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 1024 1024\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M512 1024c-69.1 0-136.2-13.5-199.3-40.2C251.7 958 197 921 150 874c-47-47-84-101.7-109.8-162.7C13.5 648.2 0 581.1 0 512c0-19.9 16.1-36 36-36s36 16.1 36 36c0 59.4 11.6 117 34.6 171.3 22.2 52.4 53.9 99.5 94.3 139.9 40.4 40.4 87.5 72.2 139.9 94.3C395 940.4 452.6 952 512 952c59.4 0 117-11.6 171.3-34.6 52.4-22.2 99.5-53.9 139.9-94.3 40.4-40.4 72.2-87.5 94.3-139.9C940.4 629 952 571.4 952 512c0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.2C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3s-13.5 136.2-40.2 199.3C958 772.3 921 827 874 874c-47 47-101.8 83.9-162.7 109.7-63.1 26.8-130.2 40.3-199.3 40.3z\\"></path></svg></div><input class=\\"progress-range\\" type=\\"range\\" min=\\"0\\" max=\\"0\\" step=\\"5\\" value=\\"0\\"><p>0:00</p></div><div class=\\"control-btns-div\\"><button class=\\"control-icons\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 16 16\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"m11.596 8.697-6.363 3.692c-.54.313-1.233-.066-1.233-.697V4.308c0-.63.692-1.01 1.233-.696l6.363 3.692a.802.802 0 0 1 0 1.393z\\"></path></svg></button><div class=\\"flex flex-row\\"><button class=\\"control-icons\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 16 16\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill-rule=\\"evenodd\\" d=\\"M8 2.81v10.38c0 .67-.81 1-1.28.53L3 10H1c-.55 0-1-.45-1-1V7c0-.55.45-1 1-1h2l3.72-3.72C7.19 1.81 8 2.14 8 2.81zm7.53 3.22l-1.06-1.06-1.97 1.97-1.97-1.97-1.06 1.06L11.44 8 9.47 9.97l1.06 1.06 1.97-1.97 1.97 1.97 1.06-1.06L13.56 8l1.97-1.97z\\"></path></svg></button><input type=\\"range\\" min=\\"0\\" max=\\"1\\" step=\\"0.01\\" value=\\"0.5\\"></div><button class=\\"control-icons\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 16 16\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill-rule=\\"evenodd\\" d=\\"M5.828 10.172a.5.5 0 0 0-.707 0l-4.096 4.096V11.5a.5.5 0 0 0-1 0v3.975a.5.5 0 0 0 .5.5H4.5a.5.5 0 0 0 0-1H1.732l4.096-4.096a.5.5 0 0 0 0-.707zm4.344 0a.5.5 0 0 1 .707 0l4.096 4.096V11.5a.5.5 0 1 1 1 0v3.975a.5.5 0 0 1-.5.5H11.5a.5.5 0 0 1 0-1h2.768l-4.096-4.096a.5.5 0 0 1 0-.707zm0-4.344a.5.5 0 0 0 .707 0l4.096-4.096V4.5a.5.5 0 1 0 1 0V.525a.5.5 0 0 0-.5-.5H11.5a.5.5 0 0 0 0 1h2.768l-4.096 4.096a.5.5 0 0 0 0 .707zm-4.344 0a.5.5 0 0 1-.707 0L1.025 1.732V4.5a.5.5 0 0 1-1 0V.525a.5.5 0 0 1 .5-.5H4.5a.5.5 0 0 1 0 1H1.732l4.096 4.096a.5.5 0 0 1 0 .707z\\"></path></svg></button><button class=\\"control-icons next-icon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M7 7v10l7-5zm9 10V7h-2v10z\\"></path></svg></button></div></div></div></div>02<div class=\\"playlist-div\\"><h2>Your Playlist</h2><button class=\\"playlist-item bg-blue-100\\">Master</button><button class=\\"playlist-item\\">File example MP4 1920 18MG</button><button class=\\"playlist-item\\">Index 0 av</button></div></div>"`;
